app/src/test/java/com/rkm/tasky/network/fakes/AuthorizationManagerFake.kt

```diff
@@ -0,0 +1,51 @@
+package com.rkm.tasky.network.fakes
+
+import com.rkm.tasky.network.authorization.abstraction.AuthorizationManager
+import com.rkm.tasky.network.authorization.model.AuthInfoDTO
+import com.rkm.tasky.network.util.NetworkError
+import com.rkm.tasky.util.result.EmptyResult
+import com.rkm.tasky.util.result.Result
+import com.rkm.tasky.util.result.asEmptyDataResult
+import com.rkm.tasky.util.result.onSuccess
+import com.rkm.tasky.util.storage.model.asAuthInfoDTO
+
+class AuthorizationManagerFake(
+    private val repository: AuthorizationRepositoryFake,
+    private val storage: SessionStorageFake
+): AuthorizationManager {
+    override suspend fun logOut(): EmptyResult<NetworkError.APIError> {
+        return repository.logoutUser().asEmptyDataResult()
+    }
+
+    override suspend fun checkAuthentication(): EmptyResult<NetworkError.APIError> {
+        return repository.checkAuthentication().asEmptyDataResult()
+    }
+
+    override suspend fun getNewAccessToken(): EmptyResult<NetworkError.APIError> {
+        val authInfo = storage.getSession() ?: return Result.Error(
+            NetworkError.APIError.UNKNOWN
+        )
+
+        val result = repository.getNewAccessToken(
+            refreshToken = authInfo.refreshToken,
+            userId = authInfo.userId
+        )
+
+        result.onSuccess { newAccessToken ->
+            println("I LOVE MEN SO MUCH <3")
```

Comment:
![awkward-black](https://github.com/user-attachments/assets/1668e553-0ff8-4fc4-a1cd-37d74ea841a6)

Reply:
I tend to use crazy println statements for debugging

Reply:
I suggest you unlearn that habit 😅 especially when in a professional environment.
Something I do to make it easier to find my debug statements is use println statements prefixed with a Marvel character name, something like this:

```
println(“Wolverine - validation success”)
println(“Deadpool - function called”)
```